sphinx/util/inspect.py

```diff
@@ -315,6 +317,76 @@ def is_builtin_class_method(obj: Any, attr_name: str) -> bool:
     return getattr(builtins, safe_getattr(cls, '__name__', '')) is cls
 
 
+def iter_args(func: Union[ast.FunctionDef, ast.AsyncFunctionDef]
+              ) -> Generator[str, None, None]:
+    """Get an iterator for arguments names from FunctionDef node."""
+    if hasattr(func.args, "posonlyargs"):  # py38 or above
+        yield from (a.arg for a in func.args.posonlyargs)  # type: ignore
+    yield from (a.arg for a in func.args.args)
+    if func.args.vararg:
+        yield func.args.vararg.arg
+    if func.args.kwarg:
+        yield func.args.kwarg.arg
+
+
+def parse_argtypes(s: str) -> Generator[str, None, None]:
+    """Parse argument part of type_comment."""
+    start = 0
+    parens = 0
+    for i, char in enumerate(s):
+        if char == '[':
+            parens += 1
+        elif char == ']':
+            parens -= 1
+        elif char == ',' and parens == 0:
```

Comment:
This doesn't correctly handle type comments like `# type: (Literal['['], int) -> int`

Reply:
You're right. I should already think about `Literal["\""]`...